x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportDeleteTrainedModelAction.java

```diff
@@ -228,6 +244,39 @@ private void deleteModel(DeleteTrainedModelAction.Request request, ClusterState
         }
     }
 
+    protected boolean modelExists(String modelId) {
+        CountDownLatch latch = new CountDownLatch(1);
+        final AtomicBoolean modelExists = new AtomicBoolean(false);
+
```

Comment:
To avoid using a latch and requiring a timeout, I think we could replace this function with an actionListener. What do you think?